### PR TITLE
feat: add accessibility Github Actions workflow

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes:

Example run with violations:
https://github.com/aws-amplify/docs/actions/runs/9276059622/job/25522208433?pr=7669

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .github/workflows/scripts/check_for_changed_pages.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      issue: { number: issue_number },
+      repo: { owner, repo }
+    } = context;
+
+    // Use the Github API to query for the list of files from the PR
+    return github
+      .paginate(
+        'GET /repos/{owner}/{repo}/pulls/{pull_number}/files',
+        { owner, repo, pull_number: issue_number },
+        (response) => response.data.filter((file) => (file.status === 'modified' || file.status === 'added'))
```
Comment (timngyn) [inline .github/workflows/scripts/check_for_changed_pages.js:18]: Could we use `async/await` here? I know one of our utilities function started using `await`, but I know we still have other scripts using the promises chaining

example of async and await from one of our scripts: 

https://github.com/aws-amplify/docs/blob/19117c75f49dcf8a81bb4c967c1ee868efa3d388/.github/workflows/scripts/utilities.js#L14-L22

File: .github/workflows/scripts/check_for_changed_pages.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          'android',
+          'angular',
+          'flutter',
+          'javascript',
+          'nextjs',
+          'react',
+          'react-native',
+          'swift',
+          'vue',
+        ]
```
Comment (timngyn) [inline .github/workflows/scripts/check_for_changed_pages.js:32]: nit: does this need a semicolon?

Comment (timngyn) [inline .github/workflows/scripts/check_for_changed_pages.js:32]: nit: should we add a semicolon? (just thinking out loud but maybe in the future we can get some linting and stuff for this folder too)

```suggestion
        ];
```

File: .github/workflows/scripts/check_for_changed_pages.js (hunk 1 above)
Comment (hbuchel) [reply .github/workflows/scripts/check_for_changed_pages.js]: I updated this in latest commit to use async/await instead (and some formatting/comment updates)

File: .github/workflows/scripts/check_for_changed_pages.js (hunk 2 above)
Comment (hbuchel) [reply .github/workflows/scripts/check_for_changed_pages.js]: Yeah, I'll add a task for that. Updated this in latest!

**Review by timngyn — APPROVED** [review]
This is awesome, great job @hbuchel!!

**Review by katiegoines — APPROVED** [review]

**Pull request merged**